I. IMPORTAMOS LAS LIBRERIAS A UTILIZAR

In [61]:
import pandas as pd
import numpy as np
import ast

II. CARGAMOS NUESTRA DATA DESDE UN ARCHIVO CSV Y VISUALIZAMOS LA DATA

In [62]:
df_movies = pd.read_csv("movies_dataset.csv", low_memory=False)

In [63]:
df_movies

adult                              belongs_to_collection    budget  \
0      False  {'id': 10194, 'name': 'Toy Story Collection', ...  30000000   
1      False                                                NaN  65000000   
2      False  {'id': 119050, 'name': 'Grumpy Old Men Collect...         0   
3      False                                                NaN  16000000   
4      False  {'id': 96871, 'name': 'Father of the Bride Col...         0   
...      ...                                                ...       ...   
45461  False                                                NaN         0   
45462  False                                                NaN         0   
45463  False                                                NaN         0   
45464  False                                                NaN         0   
45465  False                                                NaN         0   

                                                  genres  \
0      [{'id': 16, 'name': 'Animation'}, {'id': 35, '...   
1      [{'id': 12, 'name': 'Adventure'}, {'id': 14, '...   
2      [{'id': 10749, 'name': 'Romance'}, {'id': 35, ...   
3      [{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...   
4                         [{'id': 35, 'name': 'Comedy'}]   
...                                                  ...   
45461  [{'id': 18, 'name': 'Drama'}, {'id': 10751, 'n...   
45462                      [{'id': 18, 'name': 'Drama'}]   
45463  [{'id': 28, 'name': 'Action'}, {'id': 18, 'nam...   
45464                                                 []   
45465                                                 []   

                                   homepage      id    imdb_id  \
0      http://toystory.disney.com/toy-story     862  tt0114709   
1                                       NaN    8844  tt0113497   
2                                       NaN   15602  tt0113228   
3                                       NaN   31357  tt0114885   
4                                       NaN   11862  tt0113041   
...                                     ...     ...        ...   
45461  http://www.imdb.com/title/tt6209470/  439050  tt6209470   
45462                                   NaN  111109  tt2028550   
45463                                   NaN   67758  tt0303758   
45464                                   NaN  227506  tt0008536   
45465                                   NaN  461257  tt6980792   

      original_language               original_title  \
0                    en                    Toy Story   
1                    en                      Jumanji   
2                    en             Grumpier Old Men   
3                    en            Waiting to Exhale   
4                    en  Father of the Bride Part II   
...                 ...                          ...   
45461                fa                      رگ خواب   
45462                tl          Siglo ng Pagluluwal   
45463                en                     Betrayal   
45464                en          Satana likuyushchiy   
45465                en                     Queerama   

                                                overview  ... release_date  \
0      Led by Woody, Andy's toys live happily in his ...  ...   1995-10-30   
1      When siblings Judy and Peter discover an encha...  ...   1995-12-15   
2      A family wedding reignites the ancient feud be...  ...   1995-12-22   
3      Cheated on, mistreated and stepped on, the wom...  ...   1995-12-22   
4      Just when George Banks has recovered from his ...  ...   1995-02-10   
...                                                  ...  ...          ...   
45461        Rising and falling between a man and woman.  ...          NaN   
45462  An artist struggles to finish his work while a...  ...   2011-11-17   
45463  When one of her hits goes wrong, a professiona...  ...   2003-08-01   
45464  In a small town live two brothers, one a minis...  ...   1917-10-21   
45465  50 years after decriminalisati

III. HAREMOS UNA PRIMERA INSPECCIÓN SOBRE LOS METADATOS DE NUESTRO DATASET

1.- Veamos la cantidad de filas y columnas

In [64]:
df_movies.shape

(45466, 24)

2.- Veamos información sobre la data: columnas, tipos de datos, cantida de no nulos

In [65]:
df_movies.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 45466 entries, 0 to 45465
Data columns (total 24 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   adult                  45466 non-null  object 
 1   belongs_to_collection  4494 non-null   object 
 2   budget                 45466 non-null  object 
 3   genres                 45466 non-null  object 
 4   homepage               7782 non-null   object 
 5   id                     45466 non-null  object 
 6   imdb_id                45449 non-null  object 
 7   original_language      45455 non-null  object 
 8   original_title         45466 non-null  object 
 9   overview               44512 non-null  object 
 10  popularity             45461 non-null  object 
 11  poster_path            45080 non-null  object 
 12  production_companies   45463 non-null  object 
 13  production_countries   45463 non-null  object 
 14  release_date           45379 non-null  object 
 15  re

3.- Contabilizamos la cantidad de nulos por columnas

In [66]:
df_movies.isnull().sum()

adult                        0
belongs_to_collection    40972
budget                       0
genres                       0
homepage                 37684
id                           0
imdb_id                     17
original_language           11
original_title               0
overview                   954
popularity                   5
poster_path                386
production_companies         3
production_countries         3
release_date                87
revenue                      6
runtime                    263
spoken_languages             6
status                      87
tagline                  25054
title                        6
video                        6
vote_average                 6
vote_count                   6
dtype: int64

En nuestra primera observación de nuestra data podemos ver que tenemos columnas con datos anidados que debemos desglozar, ademas una cantidad importante de datos nulos en algunas columnas, y tambien debemos dar el formato correcto a los datos dependiendo del tipo de dato. A continuación procederemos a llevar a cabo las transformaciones requeridas para la limpieza de los datos de este proyecto.

IV. TRANSFORMACIONES

1.- Eliminaremos las columnas video, imdb_id, adult, original_title, vote_count, poster_path y homepage ya que no serán utilizadas.

In [67]:
df_movies.drop(columns=['video','imdb_id','adult','original_title','poster_path','homepage'], inplace=True)

In [68]:
df_movies.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 45466 entries, 0 to 45465
Data columns (total 18 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   belongs_to_collection  4494 non-null   object 
 1   budget                 45466 non-null  object 
 2   genres                 45466 non-null  object 
 3   id                     45466 non-null  object 
 4   original_language      45455 non-null  object 
 5   overview               44512 non-null  object 
 6   popularity             45461 non-null  object 
 7   production_companies   45463 non-null  object 
 8   production_countries   45463 non-null  object 
 9   release_date           45379 non-null  object 
 10  revenue                45460 non-null  float64
 11  runtime                45203 non-null  float64
 12  spoken_languages       45460 non-null  object 
 13  status                 45379 non-null  object 
 14  tagline                20412 non-null  object 
 15  ti

Como se puede observar teniamos un total de 24 columnas luego de eliminar las que no seran utilizadas ahora tenemos solo 17 columnas en nuestro dataset.

2.- Los valores nulos de los campos revenue, budget deben ser rellenados por el número 0.

In [69]:
#Contabilizamos la cantidad de nulos en nuestra columna revenue
df_movies.revenue.isnull().sum()

6

In [70]:
#Reemplazamos los valores nulos de nuestra columna revenue por ceros
df_movies['revenue'].fillna(value=0, inplace=True)

In [71]:
#Contabilizamos la cantidad de nulos en nuestra columna budget y notamos que no hay valores nulos.
df_movies.budget.isnull().sum()

0

3.- De haber fechas, deberán tener el formato aaaa-mm-dd.

In [72]:
#Visualizamos los datos de nuestra columna release_date
df_movies['release_date'].head()

0    1995-10-30
1    1995-12-15
2    1995-12-22
3    1995-12-22
4    1995-02-10
Name: release_date, dtype: object

In [73]:
#Canviamos nuestra columna a formato de tipo datetime y convertimos los valores que no cumplen el formato a nulos
df_movies['release_date'] = pd.to_datetime(df_movies['release_date'], errors='coerce')

In [74]:
df_movies.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 45466 entries, 0 to 45465
Data columns (total 18 columns):
 #   Column                 Non-Null Count  Dtype         
---  ------                 --------------  -----         
 0   belongs_to_collection  4494 non-null   object        
 1   budget                 45466 non-null  object        
 2   genres                 45466 non-null  object        
 3   id                     45466 non-null  object        
 4   original_language      45455 non-null  object        
 5   overview               44512 non-null  object        
 6   popularity             45461 non-null  object        
 7   production_companies   45463 non-null  object        
 8   production_countries   45463 non-null  object        
 9   release_date           45376 non-null  datetime64[ns]
 10  revenue                45466 non-null  float64       
 11  runtime                45203 non-null  float64       
 12  spoken_languages       45460 non-null  object        
 13  s

4.- Los valores nulos del campo release date deben eliminarse.

In [75]:
#Verificamos la cantidad de nulos en nuestra columna release_date
df_movies.release_date.isnull().sum()

90

In [76]:
#Procedemos a eliminar los valores nulos
df_movies=df_movies.dropna(subset=['release_date'])

5.- Crear la columna release_year donde extraerán el año de la fecha de estreno.

In [77]:
#Creamos la columna release_year a partir de la columna release_year solo con los años
df_movies.loc[:, 'release_year'] = df_movies['release_date'].dt.year

C:\Users\Administrador\AppData\Local\Temp\ipykernel_24432\3134557204.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_movies.loc[:, 'release_year'] = df_movies['release_date'].dt.year


In [78]:
#Cambiamos el tipo de dato a int64
df_movies.loc[:, 'release_year']=df_movies['release_year'].astype('int64')


In [79]:
#Visualizamos el tipo de dato
df_movies['release_year'].dtype

dtype('int64')

6.- Crear la columna con el retorno de inversión, llamada return con los campos revenue y budget, dividiendo estas dos últimas revenue / budget, cuando no hay datos disponibles para calcularlo, deberá tomar el valor 0.

In [80]:
#Teníamos algunos valores en la columna budget que no eran de tipo numerico por lo que procedemos a cambiarlos
df_movies['budget'] = pd.to_numeric(df_movies['budget'], errors='coerce')


C:\Users\Administrador\AppData\Local\Temp\ipykernel_24432\1718084094.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_movies['budget'] = pd.to_numeric(df_movies['budget'], errors='coerce')


In [81]:
#Cambiamos el tipo de dato de la columna budget a float
df_movies.loc[:, 'budget'] = df_movies['budget'].astype('float64')

In [82]:
df_movies['budget'].dtype

dtype('int64')

In [83]:
df_movies.info()

<class 'pandas.core.frame.DataFrame'>
Index: 45376 entries, 0 to 45465
Data columns (total 19 columns):
 #   Column                 Non-Null Count  Dtype         
---  ------                 --------------  -----         
 0   belongs_to_collection  4488 non-null   object        
 1   budget                 45376 non-null  int64         
 2   genres                 45376 non-null  object        
 3   id                     45376 non-null  object        
 4   original_language      45365 non-null  object        
 5   overview               44435 non-null  object        
 6   popularity             45376 non-null  object        
 7   production_companies   45376 non-null  object        
 8   production_countries   45376 non-null  object        
 9   release_date           45376 non-null  datetime64[ns]
 10  revenue                45376 non-null  float64       
 11  runtime                45130 non-null  float64       
 12  spoken_languages       45376 non-null  object        
 13  status

In [84]:
#Creamos la columna return como nos indica la transformación
#df_movies['budget'] = df_movies['budget'].replace(0, np.nan)
df_movies['return'] = df_movies['revenue'] / df_movies['budget'].fillna(np.inf)

C:\Users\Administrador\AppData\Local\Temp\ipykernel_24432\743583513.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_movies['return'] = df_movies['revenue'] / df_movies['budget'].fillna(np.inf)


In [85]:
#Visualizamos nuestra columna return
df_movies['return']

0        12.451801
1         4.043035
2              NaN
3         5.090760
4              inf
           ...    
45460          NaN
45462          NaN
45463          NaN
45464          NaN
45465          NaN
Name: return, Length: 45376, dtype: float64

Aqui vemos que tenemos muchos datos nulos o infinitos en nuestra columna return, el cual deberemos reemplazar por valores cero, ya que es una indicación  de la dirección del proyecto.

In [86]:
#Contabilizamos la cantidad de ceros en nuestra columna return
df_movies['return'].eq(0).sum()

3505

Aqui podemos observar que nuestra columna return tiene muchos valores de tipo nulos e infinitos. Veamos que cantidad de cada uno hay en la columna.

In [87]:
#Veamos la cantidad de valores infinitos
valoresinfinito = np.isinf(df_movies['return']).sum()
valoresinfinito


2026

In [88]:
#Veamos la cantidad de valores cero
valoresnulos = df_movies['return'].isnull().sum()
valoresnulos


34464

Como se puede observar tenemos 2026 valores infinitos y 34464 valores igual a nulos en nuestra columna return por lo que procederemos a reemplazarlos por cero.

In [89]:
#Procedemos a reemplazar los valores nulos por cero
df_movies.loc[:, 'return'] = df_movies['return'].fillna(0)


In [90]:
#Procedemos a reemplazar los valores infinitos por cero
df_movies.loc[:, 'return'] = df_movies['return'].replace(np.inf, 0)

In [91]:
#Como valor adicional comentaremos que podemos calcular la cantidad de valores cero de la siguiente forma
#valorescero = df_movies['columna1'].eq(0).sum()


In [92]:
#Veamos la información de nuestra data
df_movies.info()

<class 'pandas.core.frame.DataFrame'>
Index: 45376 entries, 0 to 45465
Data columns (total 20 columns):
 #   Column                 Non-Null Count  Dtype         
---  ------                 --------------  -----         
 0   belongs_to_collection  4488 non-null   object        
 1   budget                 45376 non-null  int64         
 2   genres                 45376 non-null  object        
 3   id                     45376 non-null  object        
 4   original_language      45365 non-null  object        
 5   overview               44435 non-null  object        
 6   popularity             45376 non-null  object        
 7   production_companies   45376 non-null  object        
 8   production_countries   45376 non-null  object        
 9   release_date           45376 non-null  datetime64[ns]
 10  revenue                45376 non-null  float64       
 11  runtime                45130 non-null  float64       
 12  spoken_languages       45376 non-null  object        
 13  status

7.- Algunos campos, como belongs_to_collection, production_companies y otros (ver diccionario de datos) están anidados, esto es o bien tienen un diccionario o una lista como valores en cada fila, ¡deberán desanidarlos para poder y unirlos al dataset de nuevo hacer alguna de las consultas de la API! O bien buscar la manera de acceder a esos datos sin desanidarlos.

In [93]:
df_movies.head()

belongs_to_collection    budget  \
0  {'id': 10194, 'name': 'Toy Story Collection', ...  30000000   
1                                                NaN  65000000   
2  {'id': 119050, 'name': 'Grumpy Old Men Collect...         0   
3                                                NaN  16000000   
4  {'id': 96871, 'name': 'Father of the Bride Col...         0   

                                              genres     id original_language  \
0  [{'id': 16, 'name': 'Animation'}, {'id': 35, '...    862                en   
1  [{'id': 12, 'name': 'Adventure'}, {'id': 14, '...   8844                en   
2  [{'id': 10749, 'name': 'Romance'}, {'id': 35, ...  15602                en   
3  [{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...  31357                en   
4                     [{'id': 35, 'name': 'Comedy'}]  11862                en   

                                            overview popularity  \
0  Led by Woody, Andy's toys live happily in his ...  21.946943   
1  When siblings Judy and Peter discover an encha...  17.015539   
2  A family wedding reignites the ancient feud be...    11.7129   
3  Cheated on, mistreated and stepped on, the wom...   3.859495   
4  Just when George Banks has recovered from his ...   8.387519   

                                production_companies  \
0     [{'name': 'Pixar Animation Studios', 'id': 3}]   
1  [{'name': 'TriStar Pictures', 'id': 559}, {'na...   
2  [{'name': 'Warner Bros.', 'id': 6194}, {'name'...   
3  [{'name': 'Twentieth Century Fox Film Corporat...   
4  [{'name': 'Sandollar Productions', 'id': 5842}...   

                                production_countries release_date  \
0  [{'iso_3166_1': 'US', 'name': 'United States o...   1995-10-30   
1  [{'iso_3166_1': 'US', 'name': 'United States o...   1995-12-15   
2  [{'iso_3166_1': 'US', 'name': 'United States o...   1995-12-22   
3  [{'iso_3166_1': 'US', 'name': 'United States o...   1995-12-22   
4  [{'iso_3166_1': 'US', 'name': 'United States o...   1995-02-10   

       revenue  runtime                                   spoken_languages  \
0  373554033.0     81.0           [{'iso_639_1': 'en', 'name': 'English'}]   
1  262797249.0    104.0  [{'iso_639_1': 'en', 'name': 'English'}, {'iso...   
2          0.0    101.0           [{'iso_639_1': 'en', 'name': 'English'}]   
3   81452156.0    127.0           [{'iso_639_1': 'en', 'name': 'English'}]   
4   76578911.0    106.0           [{'iso_639_1': 'en', 'name': 'English'}]   

     status                                            tagline  \
0  Released                                                NaN   
1  Released          Roll the dice and unleash the excitement!   
2  Released  Still Yelling. Still Fighting. Still Ready for...   
3  Released  Friends are the people who let you be yourself...   
4  Released  Just When His World Is Back To Normal... He's ...   

                         title  vote_average  vote_count  release_year  \
0                    Toy Story           7.7      5415.0          1995   
1                      Jumanji           6.9      2413.0          1995   
2             Grumpier Old Men           6.5        92.0          1995   
3            Waiting to Exhale           6.1        34.0          1995   
4  Father of the Bride Part II           5.7       173.0          1995   

      return  
0  12.451801  
1   4.043035  
2   0.000000  
3   5.090760  
4   0.000000

Como podemos observar tenemos columnas anidadas por lo que debemos definir algunas funciones para desanidar los datos contenidos en listas y diccionarios.

In [94]:
#DEFINIMOS FUNCIONES PARA DESANIDAR LAS COLUMNAS ANIDADAS

def fetch_name(obj): 
    if isinstance(obj, str) and '{' in obj:
        L=[]
        for i in ast.literal_eval(obj):
            L.append(i['name']);
        return L

In [95]:
def fetch_name2(obj): 
    if isinstance(obj, str) and '{' in obj:
        # print(obj)
        dic = ast.literal_eval(obj)
        return dic['name']

In [96]:
#APLICAMOS LAS FUNCIONES CREADAS A LAS COLUMNAS, PARA DESANIDARLAS
df_movies.loc[:, 'genres']                = df_movies['genres'].apply(fetch_name)
df_movies.loc[:, 'belongs_to_collection'] = df_movies['belongs_to_collection'].apply(fetch_name2)
df_movies.loc[:, 'production_companies']  = df_movies['production_companies'].apply(fetch_name)
df_movies.loc[:, 'production_countries']  = df_movies['production_countries'].apply(fetch_name)
df_movies.loc[:, 'spoken_languages']      = df_movies['spoken_languages'].apply(fetch_name)

In [97]:
df_movies.head(10)

belongs_to_collection    budget  \
0            Toy Story Collection  30000000   
1                            None  65000000   
2       Grumpy Old Men Collection         0   
3                            None  16000000   
4  Father of the Bride Collection         0   
5                            None  60000000   
6                            None  58000000   
7                            None         0   
8                            None  35000000   
9           James Bond Collection  58000000   

                               genres     id original_language  \
0         [Animation, Comedy, Family]    862                en   
1        [Adventure, Fantasy, Family]   8844                en   
2                   [Romance, Comedy]  15602                en   
3            [Comedy, Drama, Romance]  31357                en   
4                            [Comedy]  11862                en   
5    [Action, Crime, Drama, Thriller]    949                en   
6                   [Comedy, Romance]  11860                en   
7  [Action, Adventure, Drama, Family]  45325                en   
8       [Action, Adventure, Thriller]   9091                en   
9       [Adventure, Action, Thriller]    710                en   

                                            overview popularity  \
0  Led by Woody, Andy's toys live happily in his ...  21.946943   
1  When siblings Judy and Peter discover an encha...  17.015539   
2  A family wedding reignites the ancient feud be...    11.7129   
3  Cheated on, mistreated and stepped on, the wom...   3.859495   
4  Just when George Banks has recovered from his ...   8.387519   
5  Obsessive master thief, Neil McCauley leads a ...  17.924927   
6  An ugly duckling having undergone a remarkable...   6.677277   
7  A mischievous young boy, Tom Sawyer, witnesses...   2.561161   
8  International action superstar Jean Claude Van...    5.23158   
9  James Bond must unmask the mysterious head of ...  14.686036   

                                production_companies  \
0                          [Pixar Animation Studios]   
1  [TriStar Pictures, Teitler Film, Interscope Co...   
2                     [Warner Bros., Lancaster Gate]   
3           [Twentieth Century Fox Film Corporation]   
4       [Sandollar Productions, Touchstone Pictures]   
5  [Regency Enterprises, Forward Pass, Warner Bros.]   
6  [Paramount Pictures, Scott Rudin Productions, ...   
7                             [Walt Disney Pictures]   
8  [Universal Pictures, Imperial Entertainment, S...   
9                  [United Artists, Eon Productions]   

                         production_countries release_date      revenue  \
0                  [United States of America]   1995-10-30  373554033.0   
1                  [United States of America]   1995-12-15  262797249.0   
2                  [United States of America]   1995-12-22          0.0   
3                  [United States of America]   1995-12-22   81452156.0   
4                  [United States of America]   1995-02-10   76578911.0   
5                  [United States of America]   1995-12-15  187436818.0   
6         [Germany, United States of America]   1995-12-15          0.0   
7                  [United States of America]   1995-12-22          0.0   
8                  [United States of America]   1995-12-22   64350171.0   
9  [United Kingdom, United States of America]   1995-11-16  352194034.0   

   runtime             spoken_languages    status  \
0     81.0                    [English]  Released   
1    104.0          [English, Français]  Released   
2    101.0                    [English]  Released   
3    127.0                    [English]  Released   
4    106.0                    [English]  Released   
5    170.0           [English, Español]  Released   
6    127.0          [Français, English]  Released   
7     97.0           [English, Deutsch]  Released   
8    106.0                    [English]  Released   
9    130.0  [English, Pусский, Español]  Released   

  

In [98]:
#EXPRESAMOS LOS DATOS DE LAS COLUMNAS QUE TIENEN DATOS EN FORMA DE LISTAS A FORMA DE NO LISTAS

df_movies.loc[:, 'genres'] = df_movies['genres'].apply(lambda x: ', '.join(map(str, x)) if x is not None else None)
df_movies.loc[:, 'production_companies'] = df_movies['production_companies'].apply(lambda x: ', '.join(map(str, x)) if x is not None else None)
df_movies.loc[:, 'production_countries'] = df_movies['production_countries'].apply(lambda x: ', '.join(map(str, x)) if x is not None else None)
df_movies.loc[:, 'spoken_languages'] = df_movies['spoken_languages'].apply(lambda x: ', '.join(map(str, x)) if x is not None else None)

In [99]:
df_movies.head()

belongs_to_collection    budget                      genres  \
0            Toy Story Collection  30000000   Animation, Comedy, Family   
1                            None  65000000  Adventure, Fantasy, Family   
2       Grumpy Old Men Collection         0             Romance, Comedy   
3                            None  16000000      Comedy, Drama, Romance   
4  Father of the Bride Collection         0                      Comedy   

      id original_language                                           overview  \
0    862                en  Led by Woody, Andy's toys live happily in his ...   
1   8844                en  When siblings Judy and Peter discover an encha...   
2  15602                en  A family wedding reignites the ancient feud be...   
3  31357                en  Cheated on, mistreated and stepped on, the wom...   
4  11862                en  Just when George Banks has recovered from his ...   

  popularity                               production_companies  \
0  21.946943                            Pixar Animation Studios   
1  17.015539  TriStar Pictures, Teitler Film, Interscope Com...   
2    11.7129                       Warner Bros., Lancaster Gate   
3   3.859495             Twentieth Century Fox Film Corporation   
4   8.387519         Sandollar Productions, Touchstone Pictures   

       production_countries release_date      revenue  runtime  \
0  United States of America   1995-10-30  373554033.0     81.0   
1  United States of America   1995-12-15  262797249.0    104.0   
2  United States of America   1995-12-22          0.0    101.0   
3  United States of America   1995-12-22   81452156.0    127.0   
4  United States of America   1995-02-10   76578911.0    106.0   

    spoken_languages    status  \
0            English  Released   
1  English, Français  Released   
2            English  Released   
3            English  Released   
4            English  Released   

                                             tagline  \
0                                                NaN   
1          Roll the dice and unleash the excitement!   
2  Still Yelling. Still Fighting. Still Ready for...   
3  Friends are the people who let you be yourself...   
4  Just When His World Is Back To Normal... He's ...   

                         title  vote_average  vote_count  release_year  \
0                    Toy Story           7.7      5415.0          1995   
1                      Jumanji           6.9      2413.0          1995   
2             Grumpier Old Men           6.5        92.0          1995   
3            Waiting to Exhale           6.1        34.0          1995   
4  Father of the Bride Part II           5.7       173.0          1995   

      return  
0  12.451801  
1   4.043035  
2   0.000000  
3   5.090760  
4   0.000000

Finalmente hemos aplicado las transformaciones requeridas por la administración sin embargo aun debemos configurar la cantidad de cifras decimales de algunas columnas, además de de cambiar el tipo de dato de algunas columnas.

V. CAMBIOS FINALES

1.- Cambiaremos los valores NONE de la columna belongs_to_collection 

In [100]:
#Cambiaremos los valores None de 
df_movies.loc[:, 'belongs_to_collection'].fillna('Colección desconocida', inplace=True)

C:\Users\Administrador\AppData\Local\Temp\ipykernel_24432\374166282.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_movies.loc[:, 'belongs_to_collection'].fillna('Colección desconocida', inplace=True)


In [101]:
df_movies.head()

belongs_to_collection    budget                      genres  \
0            Toy Story Collection  30000000   Animation, Comedy, Family   
1           Colección desconocida  65000000  Adventure, Fantasy, Family   
2       Grumpy Old Men Collection         0             Romance, Comedy   
3           Colección desconocida  16000000      Comedy, Drama, Romance   
4  Father of the Bride Collection         0                      Comedy   

      id original_language                                           overview  \
0    862                en  Led by Woody, Andy's toys live happily in his ...   
1   8844                en  When siblings Judy and Peter discover an encha...   
2  15602                en  A family wedding reignites the ancient feud be...   
3  31357                en  Cheated on, mistreated and stepped on, the wom...   
4  11862                en  Just when George Banks has recovered from his ...   

  popularity                               production_companies  \
0  21.946943                            Pixar Animation Studios   
1  17.015539  TriStar Pictures, Teitler Film, Interscope Com...   
2    11.7129                       Warner Bros., Lancaster Gate   
3   3.859495             Twentieth Century Fox Film Corporation   
4   8.387519         Sandollar Productions, Touchstone Pictures   

       production_countries release_date      revenue  runtime  \
0  United States of America   1995-10-30  373554033.0     81.0   
1  United States of America   1995-12-15  262797249.0    104.0   
2  United States of America   1995-12-22          0.0    101.0   
3  United States of America   1995-12-22   81452156.0    127.0   
4  United States of America   1995-02-10   76578911.0    106.0   

    spoken_languages    status  \
0            English  Released   
1  English, Français  Released   
2            English  Released   
3            English  Released   
4            English  Released   

                                             tagline  \
0                                                NaN   
1          Roll the dice and unleash the excitement!   
2  Still Yelling. Still Fighting. Still Ready for...   
3  Friends are the people who let you be yourself...   
4  Just When His World Is Back To Normal... He's ...   

                         title  vote_average  vote_count  release_year  \
0                    Toy Story           7.7      5415.0          1995   
1                      Jumanji           6.9      2413.0          1995   
2             Grumpier Old Men           6.5        92.0          1995   
3            Waiting to Exhale           6.1        34.0          1995   
4  Father of the Bride Part II           5.7       173.0          1995   

      return  
0  12.451801  
1   4.043035  
2   0.000000  
3   5.090760  
4   0.000000

2.- Cambiaremos el tipo de dato de algunas columnas

In [102]:
#Cambiamos el tipo de dato de la columna id a int
df_movies['id']=df_movies['id'].astype('int64')


C:\Users\Administrador\AppData\Local\Temp\ipykernel_24432\1841592844.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_movies['id']=df_movies['id'].astype('int64')


In [103]:
#Cambiamos el tipo de dato de la columna popularity a float
df_movies['popularity']=df_movies['popularity'].astype('float64')


C:\Users\Administrador\AppData\Local\Temp\ipykernel_24432\3932087325.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_movies['popularity']=df_movies['popularity'].astype('float64')


In [104]:
df_movies.info()

<class 'pandas.core.frame.DataFrame'>
Index: 45376 entries, 0 to 45465
Data columns (total 20 columns):
 #   Column                 Non-Null Count  Dtype         
---  ------                 --------------  -----         
 0   belongs_to_collection  45376 non-null  object        
 1   budget                 45376 non-null  int64         
 2   genres                 42992 non-null  object        
 3   id                     45376 non-null  int64         
 4   original_language      45365 non-null  object        
 5   overview               44435 non-null  object        
 6   popularity             45376 non-null  float64       
 7   production_companies   33580 non-null  object        
 8   production_countries   39165 non-null  object        
 9   release_date           45376 non-null  datetime64[ns]
 10  revenue                45376 non-null  float64       
 11  runtime                45130 non-null  float64       
 12  spoken_languages       41608 non-null  object        
 13  status

3.- Acortaremos la cantidad de decimales de nuestras columnas popularity y return.

In [105]:
df_movies['popularity'] = df_movies['popularity'].round(2)

C:\Users\Administrador\AppData\Local\Temp\ipykernel_24432\518695597.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_movies['popularity'] = df_movies['popularity'].round(2)


In [106]:
df_movies['return'] = df_movies['return'].round(2)

C:\Users\Administrador\AppData\Local\Temp\ipykernel_24432\3742361723.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_movies['return'] = df_movies['return'].round(2)


In [107]:
#Visualizaremos nuestro dataframe final
df_movies

belongs_to_collection    budget                      genres  \
0                Toy Story Collection  30000000   Animation, Comedy, Family   
1               Colección desconocida  65000000  Adventure, Fantasy, Family   
2           Grumpy Old Men Collection         0             Romance, Comedy   
3               Colección desconocida  16000000      Comedy, Drama, Romance   
4      Father of the Bride Collection         0                      Comedy   
...                               ...       ...                         ...   
45460           Colección desconocida         0      Drama, Action, Romance   
45462           Colección desconocida         0                       Drama   
45463           Colección desconocida         0     Action, Drama, Thriller   
45464           Colección desconocida         0                        None   
45465           Colección desconocida         0                        None   

           id original_language  \
0         862                en   
1        8844                en   
2       15602                en   
3       31357                en   
4       11862                en   
...       ...               ...   
45460   30840                en   
45462  111109                tl   
45463   67758                en   
45464  227506                en   
45465  461257                en   

                                                overview  popularity  \
0      Led by Woody, Andy's toys live happily in his ...       21.95   
1      When siblings Judy and Peter discover an encha...       17.02   
2      A family wedding reignites the ancient feud be...       11.71   
3      Cheated on, mistreated and stepped on, the wom...        3.86   
4      Just when George Banks has recovered from his ...        8.39   
...                                                  ...         ...   
45460  Yet another version of the classic epic, with ...        5.68   
45462  An artist struggles to finish his work while a...        0.18   
45463  When one of her hits goes wrong, a professiona...        0.90   
45464  In a small town live two brothers, one a minis...        0.00   
45465  50 years after decriminalisation of homosexual...        0.16   

                                    production_companies  \
0                                Pixar Animation Studios   
1      TriStar Pictures, Teitler Film, Interscope Com...   
2                           Warner Bros., Lancaster Gate   
3                 Twentieth Century Fox Film Corporation   
4             Sandollar Productions, Touchstone Pictures   
...                                                  ...   
45460  Westdeutscher Rundfunk (WDR), Working Title Fi...   
45462                                        Sine Olivia   
45463                            American World Pictures   
45464                                          Yermoliev   
45465                                               None   

                                    production_countries release_date  \
0                               United States of America   1995-10-30   
1                               United States of America   1995-12-15   
2                               United States of America   1995-12-22   
3                               United States of America   1995-12-22   
4                               United States of America   1995-02-10   
...                                                  ...          ...   
45460  Canada, Germany, United Kingdom, United States...   1991-05-13   
45462                                        Philippines   2011-11-17   
45463                           United States of America   2003-08-01   
45464                                             Russia   1917-10-21   
45465                                     United Kingdom   2017-06-09   

           revenue  runtime   spoken_languages    status  \
0      373554033.0     81.0            English  Released   
1      262797249.0    104.0  English, Français  Released   
2  

In [108]:
#Visualizamos finalmente la información de nuestro dataframe
df_movies.info()

<class 'pandas.core.frame.DataFrame'>
Index: 45376 entries, 0 to 45465
Data columns (total 20 columns):
 #   Column                 Non-Null Count  Dtype         
---  ------                 --------------  -----         
 0   belongs_to_collection  45376 non-null  object        
 1   budget                 45376 non-null  int64         
 2   genres                 42992 non-null  object        
 3   id                     45376 non-null  int64         
 4   original_language      45365 non-null  object        
 5   overview               44435 non-null  object        
 6   popularity             45376 non-null  float64       
 7   production_companies   33580 non-null  object        
 8   production_countries   39165 non-null  object        
 9   release_date           45376 non-null  datetime64[ns]
 10  revenue                45376 non-null  float64       
 11  runtime                45130 non-null  float64       
 12  spoken_languages       41608 non-null  object        
 13  status

VI. EXPORTAMOS NUESTRO DATAFRAME FINAL

In [109]:
df_movies.to_csv('data_movies.csv', index=False)

VII. VAMOS A IMPORTAR OTRA DATA A PARTIR DEL ARCHIVO CREDITS.CSV

In [110]:
df_credits = pd.read_csv("credits.csv", low_memory=False)

In [111]:
df_credits

cast  \
0      [{'cast_id': 14, 'character': 'Woody (voice)',...   
1      [{'cast_id': 1, 'character': 'Alan Parrish', '...   
2      [{'cast_id': 2, 'character': 'Max Goldman', 'c...   
3      [{'cast_id': 1, 'character': "Savannah 'Vannah...   
4      [{'cast_id': 1, 'character': 'George Banks', '...   
...                                                  ...   
45471  [{'cast_id': 0, 'character': '', 'credit_id': ...   
45472  [{'cast_id': 1002, 'character': 'Sister Angela...   
45473  [{'cast_id': 6, 'character': 'Emily Shaw', 'cr...   
45474  [{'cast_id': 2, 'character': '', 'credit_id': ...   
45475                                                 []   

                                                    crew      id  
0      [{'credit_id': '52fe4284c3a36847f8024f49', 'de...     862  
1      [{'credit_id': '52fe44bfc3a36847f80a7cd1', 'de...    8844  
2      [{'credit_id': '52fe466a9251416c75077a89', 'de...   15602  
3      [{'credit_id': '52fe44779251416c91011acb', 'de...   31357  
4      [{'credit_id': '52fe44959251416c75039ed7', 'de...   11862  
...                                                  ...     ...  
45471  [{'credit_id': '5894a97d925141426c00818c', 'de...  439050  
45472  [{'credit_id': '52fe4af1c3a36847f81e9b15', 'de...  111109  
45473  [{'credit_id': '52fe4776c3a368484e0c8387', 'de...   67758  
45474  [{'credit_id': '533bccebc3a36844cf0011a7', 'de...  227506  
45475  [{'credit_id': '593e676c92514105b702e68e', 'de...  461257  

[45476 rows x 3 columns]

VIII. UNIENDO NUESTROS DATAFRAMES

In [112]:
df_unido = pd.merge(df_movies, df_credits, on='id', how='inner')

In [113]:
df_unido

belongs_to_collection    budget                      genres  \
0                Toy Story Collection  30000000   Animation, Comedy, Family   
1               Colección desconocida  65000000  Adventure, Fantasy, Family   
2           Grumpy Old Men Collection         0             Romance, Comedy   
3               Colección desconocida  16000000      Comedy, Drama, Romance   
4      Father of the Bride Collection         0                      Comedy   
...                               ...       ...                         ...   
45446           Colección desconocida         0      Drama, Action, Romance   
45447           Colección desconocida         0                       Drama   
45448           Colección desconocida         0     Action, Drama, Thriller   
45449           Colección desconocida         0                        None   
45450           Colección desconocida         0                        None   

           id original_language  \
0         862                en   
1        8844                en   
2       15602                en   
3       31357                en   
4       11862                en   
...       ...               ...   
45446   30840                en   
45447  111109                tl   
45448   67758                en   
45449  227506                en   
45450  461257                en   

                                                overview  popularity  \
0      Led by Woody, Andy's toys live happily in his ...       21.95   
1      When siblings Judy and Peter discover an encha...       17.02   
2      A family wedding reignites the ancient feud be...       11.71   
3      Cheated on, mistreated and stepped on, the wom...        3.86   
4      Just when George Banks has recovered from his ...        8.39   
...                                                  ...         ...   
45446  Yet another version of the classic epic, with ...        5.68   
45447  An artist struggles to finish his work while a...        0.18   
45448  When one of her hits goes wrong, a professiona...        0.90   
45449  In a small town live two brothers, one a minis...        0.00   
45450  50 years after decriminalisation of homosexual...        0.16   

                                    production_companies  \
0                                Pixar Animation Studios   
1      TriStar Pictures, Teitler Film, Interscope Com...   
2                           Warner Bros., Lancaster Gate   
3                 Twentieth Century Fox Film Corporation   
4             Sandollar Productions, Touchstone Pictures   
...                                                  ...   
45446  Westdeutscher Rundfunk (WDR), Working Title Fi...   
45447                                        Sine Olivia   
45448                            American World Pictures   
45449                                          Yermoliev   
45450                                               None   

                                    production_countries release_date  ...  \
0                               United States of America   1995-10-30  ...   
1                               United States of America   1995-12-15  ...   
2                               United States of America   1995-12-22  ...   
3                               United States of America   1995-12-22  ...   
4                               United States of America   1995-02-10  ...   
...                                                  ...          ...  ...   
45446  Canada, Germany, United Kingdom, United States...   1991-05-13  ...   
45447                                        Philippines   2011-11-17  ...   
45448                           United States of America   2003-08-01  ...   
45449                                             Russia   1917-10-21  ...   
45450                                     United Kingdom   2017-06-09  ...   

        spoken_languages    status  \
0                English  Released   
1      English, Français  Released   
2        

IX. DESANIDAMOS LAS COLUMNAS CAST Y CREW

In [114]:
#Aplicamos nuestra funcion fetch_name para extraer los nombres de los actores de la pelicula
df_unido['cast'] = df_unido['cast'].apply(fetch_name)

In [115]:
df_unido

belongs_to_collection    budget                      genres  \
0                Toy Story Collection  30000000   Animation, Comedy, Family   
1               Colección desconocida  65000000  Adventure, Fantasy, Family   
2           Grumpy Old Men Collection         0             Romance, Comedy   
3               Colección desconocida  16000000      Comedy, Drama, Romance   
4      Father of the Bride Collection         0                      Comedy   
...                               ...       ...                         ...   
45446           Colección desconocida         0      Drama, Action, Romance   
45447           Colección desconocida         0                       Drama   
45448           Colección desconocida         0     Action, Drama, Thriller   
45449           Colección desconocida         0                        None   
45450           Colección desconocida         0                        None   

           id original_language  \
0         862                en   
1        8844                en   
2       15602                en   
3       31357                en   
4       11862                en   
...       ...               ...   
45446   30840                en   
45447  111109                tl   
45448   67758                en   
45449  227506                en   
45450  461257                en   

                                                overview  popularity  \
0      Led by Woody, Andy's toys live happily in his ...       21.95   
1      When siblings Judy and Peter discover an encha...       17.02   
2      A family wedding reignites the ancient feud be...       11.71   
3      Cheated on, mistreated and stepped on, the wom...        3.86   
4      Just when George Banks has recovered from his ...        8.39   
...                                                  ...         ...   
45446  Yet another version of the classic epic, with ...        5.68   
45447  An artist struggles to finish his work while a...        0.18   
45448  When one of her hits goes wrong, a professiona...        0.90   
45449  In a small town live two brothers, one a minis...        0.00   
45450  50 years after decriminalisation of homosexual...        0.16   

                                    production_companies  \
0                                Pixar Animation Studios   
1      TriStar Pictures, Teitler Film, Interscope Com...   
2                           Warner Bros., Lancaster Gate   
3                 Twentieth Century Fox Film Corporation   
4             Sandollar Productions, Touchstone Pictures   
...                                                  ...   
45446  Westdeutscher Rundfunk (WDR), Working Title Fi...   
45447                                        Sine Olivia   
45448                            American World Pictures   
45449                                          Yermoliev   
45450                                               None   

                                    production_countries release_date  ...  \
0                               United States of America   1995-10-30  ...   
1                               United States of America   1995-12-15  ...   
2                               United States of America   1995-12-22  ...   
3                               United States of America   1995-12-22  ...   
4                               United States of America   1995-02-10  ...   
...                                                  ...          ...  ...   
45446  Canada, Germany, United Kingdom, United States...   1991-05-13  ...   
45447                                        Philippines   2011-11-17  ...   
45448                           United States of America   2003-08-01  ...   
45449                                             Russia   1917-10-21  ...   
45450                                     United Kingdom   2017-06-09  ...   

        spoken_languages    status  \
0                English  Released   
1      English, Français  Released   
2        

Para extraer el nombre del director de la pelicula de nuestra columna crew vamos a crear una nueva funcion llamada fetch_director

In [116]:
#Creamos la función que nos ayudara a extraer el nombre del director verificando el empleo job igual a director
def fetch_director(obj):
    L = []
    for i in ast.literal_eval(obj):
        if i['job'] == 'Director':
            L.append(i['name'])
    return L

In [117]:
#Aplicamos nuestra funcion a la columna crew
df_unido['crew']=df_unido['crew'].apply(fetch_director)

In [118]:
#Cambiaremos el nombre de nuestra columna crew a director
df_unido = df_unido.rename(columns={'crew': 'director'})


In [119]:
#Visualizamos nuestra data
df_unido.head()

belongs_to_collection    budget                      genres  \
0            Toy Story Collection  30000000   Animation, Comedy, Family   
1           Colección desconocida  65000000  Adventure, Fantasy, Family   
2       Grumpy Old Men Collection         0             Romance, Comedy   
3           Colección desconocida  16000000      Comedy, Drama, Romance   
4  Father of the Bride Collection         0                      Comedy   

      id original_language                                           overview  \
0    862                en  Led by Woody, Andy's toys live happily in his ...   
1   8844                en  When siblings Judy and Peter discover an encha...   
2  15602                en  A family wedding reignites the ancient feud be...   
3  31357                en  Cheated on, mistreated and stepped on, the wom...   
4  11862                en  Just when George Banks has recovered from his ...   

   popularity                               production_companies  \
0       21.95                            Pixar Animation Studios   
1       17.02  TriStar Pictures, Teitler Film, Interscope Com...   
2       11.71                       Warner Bros., Lancaster Gate   
3        3.86             Twentieth Century Fox Film Corporation   
4        8.39         Sandollar Productions, Touchstone Pictures   

       production_countries release_date  ...   spoken_languages    status  \
0  United States of America   1995-10-30  ...            English  Released   
1  United States of America   1995-12-15  ...  English, Français  Released   
2  United States of America   1995-12-22  ...            English  Released   
3  United States of America   1995-12-22  ...            English  Released   
4  United States of America   1995-02-10  ...            English  Released   

                                             tagline  \
0                                                NaN   
1          Roll the dice and unleash the excitement!   
2  Still Yelling. Still Fighting. Still Ready for...   
3  Friends are the people who let you be yourself...   
4  Just When His World Is Back To Normal... He's ...   

                         title vote_average vote_count  release_year  return  \
0                    Toy Story          7.7     5415.0          1995   12.45   
1                      Jumanji          6.9     2413.0          1995    4.04   
2             Grumpier Old Men          6.5       92.0          1995    0.00   
3            Waiting to Exhale          6.1       34.0          1995    5.09   
4  Father of the Bride Part II          5.7      173.0          1995    0.00   

                                                cast           director  
0  [Tom Hanks, Tim Allen, Don Rickles, Jim Varney...    [John Lasseter]  
1  [Robin Williams, Jonathan Hyde, Kirsten Dunst,...     [Joe Johnston]  
2  [Walter Matthau, Jack Lemmon, Ann-Margret, Sop...    [Howard Deutch]  
3  [Whitney Houston, Angela Bassett, Loretta Devi...  [Forest Whitaker]  
4  [Steve Martin, Diane Keaton, Martin Short, Kim...    [Charles Shyer]  

[5 rows x 22 columns]

Aqui podemos ver que los datos tanto de la columna cast como de la columna director estan aun en forma de listas por lo que procederemos a extraer esos valores en forma de no listas

In [120]:
#Cambiamos a no listas las filas de la columna cast
df_unido.loc[:, 'cast'] = df_unido['cast'].apply(lambda x: ', '.join(map(str, x)) if x is not None else None)


In [121]:
#Cambiamos a no listas las filas de la columna movie_director
df_unido.loc[:, 'director'] = df_unido['director'].apply(lambda x: ', '.join(map(str, x)) if x is not None else None)


In [122]:
df_unido.head()

belongs_to_collection    budget                      genres  \
0            Toy Story Collection  30000000   Animation, Comedy, Family   
1           Colección desconocida  65000000  Adventure, Fantasy, Family   
2       Grumpy Old Men Collection         0             Romance, Comedy   
3           Colección desconocida  16000000      Comedy, Drama, Romance   
4  Father of the Bride Collection         0                      Comedy   

      id original_language                                           overview  \
0    862                en  Led by Woody, Andy's toys live happily in his ...   
1   8844                en  When siblings Judy and Peter discover an encha...   
2  15602                en  A family wedding reignites the ancient feud be...   
3  31357                en  Cheated on, mistreated and stepped on, the wom...   
4  11862                en  Just when George Banks has recovered from his ...   

   popularity                               production_companies  \
0       21.95                            Pixar Animation Studios   
1       17.02  TriStar Pictures, Teitler Film, Interscope Com...   
2       11.71                       Warner Bros., Lancaster Gate   
3        3.86             Twentieth Century Fox Film Corporation   
4        8.39         Sandollar Productions, Touchstone Pictures   

       production_countries release_date  ...   spoken_languages    status  \
0  United States of America   1995-10-30  ...            English  Released   
1  United States of America   1995-12-15  ...  English, Français  Released   
2  United States of America   1995-12-22  ...            English  Released   
3  United States of America   1995-12-22  ...            English  Released   
4  United States of America   1995-02-10  ...            English  Released   

                                             tagline  \
0                                                NaN   
1          Roll the dice and unleash the excitement!   
2  Still Yelling. Still Fighting. Still Ready for...   
3  Friends are the people who let you be yourself...   
4  Just When His World Is Back To Normal... He's ...   

                         title vote_average vote_count  release_year  return  \
0                    Toy Story          7.7     5415.0          1995   12.45   
1                      Jumanji          6.9     2413.0          1995    4.04   
2             Grumpier Old Men          6.5       92.0          1995    0.00   
3            Waiting to Exhale          6.1       34.0          1995    5.09   
4  Father of the Bride Part II          5.7      173.0          1995    0.00   

                                                cast         director  
0  Tom Hanks, Tim Allen, Don Rickles, Jim Varney,...    John Lasseter  
1  Robin Williams, Jonathan Hyde, Kirsten Dunst, ...     Joe Johnston  
2  Walter Matthau, Jack Lemmon, Ann-Margret, Soph...    Howard Deutch  
3  Whitney Houston, Angela Bassett, Loretta Devin...  Forest Whitaker  
4  Steve Martin, Diane Keaton, Martin Short, Kimb...    Charles Shyer  

[5 rows x 22 columns]

In [123]:
#Visualizamos la información de nuestro dataset final
df_unido.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 45451 entries, 0 to 45450
Data columns (total 22 columns):
 #   Column                 Non-Null Count  Dtype         
---  ------                 --------------  -----         
 0   belongs_to_collection  45451 non-null  object        
 1   budget                 45451 non-null  int64         
 2   genres                 43067 non-null  object        
 3   id                     45451 non-null  int64         
 4   original_language      45440 non-null  object        
 5   overview               44510 non-null  object        
 6   popularity             45451 non-null  float64       
 7   production_companies   33634 non-null  object        
 8   production_countries   39232 non-null  object        
 9   release_date           45451 non-null  datetime64[ns]
 10  revenue                45451 non-null  float64       
 11  runtime                45205 non-null  float64       
 12  spoken_languages       41677 non-null  object        
 13  s

X. IMPORTAMOS NUESTRA DATA FINAL

In [124]:
df_unido.to_csv('data_movies_completa.csv', index=False)